src/clj/rems/application/commands.clj

```diff
@@ -263,6 +263,13 @@
 (defn- ok [& events]
   (ok-with-data nil events))
 
+(defn- build-forms-list [catalogue-item-ids {:keys [get-catalogue-item]}]
+  (->> catalogue-item-ids
+       (mapv get-catalogue-item)
+       (mapv :formid)
+       (distinct)
```

Comment:
had to check the source – distinct keeps the first occurrence of each value so this should give us a stable ordering (as long as catalogue-item-ids is stable)